src/printer.js

```diff
@@ -123,7 +123,10 @@ function genericPrintNoParens(path, options, print) {
             if (
               util.hasNewline(
                 options.originalText,
-                util.locEnd(childPath.getValue())
+                util.skipNewline(
```

Comment:
You probably want `skipToLineEnd`. That makes sure to skip any thing that might be after the expression, like spaces or semicolons, and also skips the newline at the end of it.